data-prepper-api/src/test/java/org/opensearch/dataprepper/model/event/JacksonEventTest.java

```diff
@@ -456,6 +456,33 @@ public void testBuild_withStringData() {
         assertThat(event.get("foo", String.class), is(equalTo("bar")));
     }
 
+    @Test
```

Comment:
Please also add some test cases with non-String values in the Event. You will probably find that these fail because of the code which casts strings. See my comment above.